docs/cli/dotnet-test.md

```diff
@@ -0,0 +1,137 @@
+        "dotnet-test-xunit": "1.0.0-*"
+    },
+    "testRunner": "xunit",
+
+    "frameworks": {
+        "netstandard1.5": {
+        }
+    }
+}
+```
+Dotnet test supports two running modes:
+
+1. Console: In console mode, dotnet test simply executes fully whatever command gets passed to it and outputs the results. Anytime you
+invoke dotnet test without passing --port, it will run in console mode, which in turn will cause the runner to run in console mode.
+2. Design time: Anytime you pass a port to dotnet test, we will run in design time. That means that dotnet test will connect to that port
+using TCP and will then exchange a established set of messages with whatever else is connected to that port. When this happens, the runner
+also receives a port (a new one, mind you) that dotnet test will use to communicate with it. The reason why the runner also uses TCP to
+communicate with dotnet test is because in design mode, it is not sufficient to just output results to the console. We need to send the
+adapter structure messages containing the results of the test execution.
+
+### Communication protocol in design time.
+
+1. Because during design time, dotnet test connects to a port when it starts up, the adapter needs to be listening on 
+that port otherwise dotnet test will fail. We did it like this so that the adapter could reserve all the ports it needs 
+by binding and listening to them before dotnet test ran and tried to get ports for the runner.
+2. Once dotnet test starts, it sends a TestSession.Connected message to the adapter indicating that it is ready to receive messages.
+3. It is possible to send an optional 
+[version check](https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.Extensions.Testing.Abstractions/Messages/ProtocolVersionMessage.cs) 
+message with the adapter version of the protocol in it. Dotnet test will send back the version of the protocol that it supports.
+
+All messages have the format described here: 
+[Message.cs](https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.Extensions.Testing.Abstractions/Messages/Message.cs). 
+The payload formats for each message is described in links to the classes used to de/serialize the information in the description of the protocol.
+
+#### Test Execution
+![alt tag](../images/DotnetTestExecuteTests.png)
+
+1. After the optional version check, the adapter sends a TestExecution.GetTestRunnerProcessStartInfo, with the 
+[tests](https://github.com/dotnet/cli/blob/rel/1.0.0/src/Microsoft.Extensions.Testing.Abstractions/Messages/RunTestsMessage.cs) it wants to execute inside of it. Dotnet test sends back a FileName and Arguments inside a [TestStartInfo](https://github.com/dotnet/cli/blob/rel/1.0.0/src/dotnet/commands/dotnet-test/TestStartInfo.cs) payload that the adapter can use to start the runner. In the past, we would send the list of tests to run as part of that argument, but we were actually going over the command line size limit for some test projects.
```

Comment:
I'd get rid of these comments about the design (In the past, we would send the ..."  and also the earlier comments about "We did it like this..."  to avoid distraction with unnecessary background information.